blosc/blosc2.c

```diff
@@ -1166,8 +1166,13 @@ static int serial_blosc(struct thread_context* thread_context) {
       }
       else {
         /* Regular decompression */
+        int32_t boffset = sw32_(bstarts + j);
+        if (boffset <= 0 || boffset >= context->srcsize) {
+          /* Invalid block offset encountered */
+          return -1;
+        }
         cbytes = blosc_d(thread_context, bsize, leftoverblock,
-                         context->src + sw32_(bstarts + j), context->srcsize,
+                         context->src + boffset, context->srcsize - boffset,
```

Comment:
I guess that you need to pass `context->srcsize - boffset` instead of the original `context->srcsize` for security.  However, in order to do that, you need to know the size of the compressed block which is not `context->srcsize - boffset`.  I think the only way to guess that size is to use something like `sw32_(bstarts + j + 1) - sw32_(bstarts + j)`.  Have into account that when `j > nblocks` you rather need to use `context->srcsize - sw32_(bstarts + j)`.